libsolidity/analysis/GlobalContext.h

```diff
@@ -57,6 +59,84 @@ class GlobalContext
 	std::vector<Declaration const*> declarations() const;
 
 private:
+	/// Magic variables get negative ids for easy differentiation
+	inline int magicVariableToID(std::string const& _name) const
```

Comment:
You don't have to move implementation to the header to make it a member of the class. It would be better to leave it where it was, just put the declaration here. And then `magicVariableToID()` does not have to be a member at all (otherwise it should have been a `static` member anyway).

Reply:
Oh, no. It was because I made it `inline`. It needs to be a member access the `m_evmVersion`, of course I can also pass it as parameter, but then I would need to change `GlobalContext::currentThis()` and `GlobalContext::currentSupper()` to pass `m_evmVersion` as parameter to `magicVariableToID` as well. So it was just a lazy way to make fewer changes in the code haha.

Reply:
Ops, actually you are right. Just noticed that I'm not using `m_evmVersion` in `magicVariableToID` anymore. Before I was doing it, so `blobhash` would only get a ID if `m_evmversion >= cancun` but as this could led to some problems in the future when more IDs get added, I decided to revert that but kept the other changes. I will fix it. Thanks ;)

Reply:
Looking at the code now, it would also make sense to keep `constructMagicVariables()` out of the class and just add the `EVMVersion` parameter to it. You would not have to store the value in the class then as the function is invoked only once and then we don't need the version for anything. It's not a big difference so it's also fine to leave it as is, but the code would be more straightforward that way.